shared/reducers/favorite.js

```diff
@@ -2,18 +2,44 @@
 
 import * as Constants from '../constants/favorite'
 import * as CommonConstants from '../constants/common'
-
+import _ from 'lodash'
 import type {FavoriteAction} from '../constants/favorite'
-import type {Folder} from '../constants/types/flow-types'
+import type {Props} from '../folders/render'
 
 type State = {
-  folders: ?Array<Folder>
+  folders: ?Props
 }
 
 const initialState = {
   folders: null
 }
 
+const folderToProps = (folders, username) => {
```

Comment:
convert what we get into what we need. the backend doesn't currently give us the individual users, which we'll need eventually